src/sentry/rules/history/preview.py

```diff
@@ -205,13 +210,19 @@ def get_top_groups(
     end: datetime,
     condition_activity: GroupActivityMap,
     dataset_map: Dict[int, Dataset],
+    has_issue_state_condition: bool = True,
 ) -> GroupActivityMap:
     """
-    Filters the activity to contain only groups that have the most events in the past 2 weeks.
+    Filters the activity to contain only groups that have the most events (out of the given groups) in the past 2 weeks.
+    If no groups are provided because there are no issue state change conditions, returns the top groups overall.
 
     Since frequency conditions require one snuba query per groups, we need to limit the number groups we process.
     """
     datasets = {dataset_map.get(group) for group in condition_activity.keys()}
+    if not has_issue_state_condition:
```

Comment:
can we infer this from the empty dataset instead of needing the extra boolean passed here?

Reply:
It's possible that the dataset can be empty and we also have issue state conditions, this just means there were no issue state activities and in this case we shouldn't be creating activity from freq conditions